lib/galaxy/config/__init__.py

```diff
@@ -861,6 +872,22 @@ def _process_config(self, kwargs):
                 'filters': ['stack']
             }
 
+    def _configure_dataset_storage(self):
+        # The default for `file_path` has changed in 20.09; we may need to fall back to the old default
+        self._set_alt_paths('file_path', self._in_data_dir('files'))  # this is called BEFORE guessing id/uuid
+        ID, UUID = 'id', 'uuid'
+        if self.is_set('object_store_store_by'):
+            assert self.object_store_store_by in [ID, UUID], f'Invalid value for object_store_store_by [{self.object_store_store_by}]'
```

Comment:
```suggestion
            assert self.object_store_store_by in [ID, UUID], "Invalid value for object_store_store_by [{}]".format(self.object_store_store_by)
```

Reply:
Committing for now, but I may have to revert these more recent commits dealing with uuid/id